src/azure-cli/azure/cli/command_modules/redis/custom.py

```diff
@@ -125,7 +125,21 @@ def cli_redis_create_server_link(cmd, client, resource_group_name, name, server_
     params = RedisLinkedServerCreateParameters(linked_redis_cache_id=cache_to_link.id,
                                                linked_redis_cache_location=cache_to_link.location,
                                                server_role=replication_role)
-    return client.create(resource_group_name, name, cache_to_link.name, params)
+    return client.begin_create(resource_group_name, name, cache_to_link.name, params)
+
+
+def cli_redis_patch_schedule_create_or_update(client, resource_group_name, name, schedule_entries):
+    from azure.mgmt.redis.models import RedisPatchSchedule
+    param = RedisPatchSchedule(schedule_entries=schedule_entries)
+    return client.create_or_update(resource_group_name, name, "default", param)
+
+
+def cli_redis_patch_schedule_get(client, resource_group_name, name):
```

Comment:
Do we need to make these changes too? Ideally if we just define it with the right factory in commands.py, it will auto get from the SDK without a redirection here. Redisrection is required only if you want to do special cases. like first getting a Resource group before passing the name etc

Reply:
The reason for redirection is to avoid a new required param "default".
The new SDK has a mandatory parameter "default" and it can take only one value "default".
I think it is the result of adding the new param in our api spec.
To avoid this mandatory parameter in CLI and to ensure backward compatibility, we have this redirection with that param included implicitly.